exasol_sphinx_github_pages_generator/deploy_github_pages.py

```diff
@@ -1,177 +1,41 @@
-        print(f"committing changes because changes_exist is {changes_exists.returncode}")
-        run(["git", "commit", "--no-verify", "-m",
-             f"Update documentation from source branch {source_branch} with commit id {current_commit_id}"], check=True)
-        if push_origin != "" and push_enabled == "push":
-            print(f"Git push {push_origin} {target_branch}")
-            run(["git", "push", push_origin, target_branch], check=True)
-    elif 0 == changes_exists.returncode:
-        print("No changes to commit.")
-    else:
-        print('A Error occurred in run(["git", "diff-index", "--quiet", "HEAD", "--"], capture_output=True, text=True)')
-        print(f" changes_exists.returncode: {changes_exists.returncode}")
-        print(f" changes_exists.stderr: {changes_exists.stderr}")
-        print(f" changes_exists.stdout: {changes_exists.stdout}")
-    os.chdir(currentworkdir)
-
-
-def clean_worktree(worktree):
-    wt = Path(worktree)
-    if wt.exists():
-        print(f"Cleanup git worktree {worktree}")
-        run(["git", "worktree", "remove", "--force", worktree], check=True)
+import sys
+from inspect import cleandoc
 
 
 def deploy_github_pages(argv):
     args = Parser(argv).args
-
+    original_workdir = os.getcwd()
     source_dir = args.source_dir
-    print("Commandline parameter")
-    print("source_dir= " + source_dir)
-    print("module_path= " + str(args.module_path))
-    print("TARGET_BRANCH=" + args.target_branch)
-    print("PUSH_ORIGIN=" + args.push_origin)
-    print("PUSH_ENABLED=" + args.push_enabled)
-    print("SOURCE_BRANCH=" + args.source_branch + "\n")
-
     current_commit_id = run(["git", "rev-parse", "HEAD"], capture_output=True, text=True, check=True)
-    with TemporaryDirectory() as tempdir:
-        worktree = tempdir + "/worktree"
-        build_dir = tempdir + "/build"
-        intermediate_dir = tempdir + "/intermediate"
-        os.mkdir(build_dir)
+    print(cleandoc("Commandline parameter \n"
+                   f"source_dir= {source_dir} \n"
+                   f"module_path= {str(args.module_path)} \n"
+                   f"TARGET_BRANCH= {args.target_branch} \n"
+                   f"PUSH_ORIGIN= {args.push_origin} \n"
+                   f"PUSH_ENABLED= {args.push_enabled} \n"
+                   f"SOURCE_BRANCH= {args.source_branch} \n"))
 
-        print("Using following Directories:")
-        print("TMP=" + tempdir)
-        print("WORKTREE=" + worktree)
-        print("BUILD_DIR=" + build_dir)
-        print("CURRENT_COMMIT_ID=" + current_commit_id.stdout + "\n")
+    with TemporaryDirectory() as tempdir:
+        deployer = GithubPagesDeployer(source_dir, args.source_branch, current_commit_id, args.module_path,
+                                       args.target_branch, args.push_origin, args.push_enabled,
+                                       tempdir)
+        os.mkdir(deployer.build_dir)
+        print(cleandoc(f"Using following Directories: \n"
```

Comment:
cleandoc is only needed to clean up, make sure indentation etc. is correct if multi line strings are used, so you would write it like: